### PR TITLE
Add specification for fees communication

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

**Review by MavenRain — APPROVED** [review]

**Comment by MavenRain** [comment]
It looks like the rustc version needs to be updated on the repo, but looks good otherwise.

File: attributes/network/cddl/15_fee.cddl [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    ; The entry that triggered the fee. This is for reference only.
+    ; The actual calculation is not shared here. Since events are in
+    ; chronological order, they can be used to reconstruct the fee table
+    ; at this point in time.
+    ; Since multiple costs might be incurred, this contains all fee
+    ; table entries that were used in calculation of the fee.
+    3 => [ *fee-table-entry-id ],
+
+    ; Determined by the neighborhood. This is to give some reason for the fee,
+    ; in human and/or machine readable format. It is optional.
```
Comment (fmorency) [inline attributes/network/cddl/15_fee.cddl:52]:
```suggestion
    ; in human and/or machine readable format.
```

**Review by fmorency — CHANGES_REQUESTED** [review]
The first round of comments. I'm still not done. I'll finish tomorrow morning.

File: attributes/network/9_account_features/15_account_fee.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+== Fee Account Feature Attribute (#15)
+:cddl: ./cddl/
+
+This attribute adds support for paying fees using this account.
+If this attribute is present in an account, any allowed (see <<Role>>) identities can specify this account in the fee request attribute.
+
+== Role
+A new role is added by this feature, `"canDeductFeeFrom"`, which allow any identities to perform regular ledger transactions.
+Identities MUST either be an `"owner"` or have this role to be allowed to use this account to pay for requests.
```
[the remark this answers is not in the document]
Comment (hansl) [reply attributes/network/9_account_features/15_account_fee.adoc:9]: It becomes muddy when we want to add payment for another's account on anything else. I'll clarify, but it's gonna get complex really fast.

[the remark this answers is not in the document]
Comment (hansl) [reply attributes/network/9_account_features/15_account_fee.adoc:9]: Basically, anything that's not a request should be owner paid. At least for now.

File: attributes/network/15_fee.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Fees are cumulative.
+For example, a storage neighborhood may charge 1 token per storage call, and 1 token per month per gigabytes for storage, and 1 token per week per kilobytes of bandwidth taken.
+
+== Fee Payment
+
+Fees can be charged for various reasons;
+
+1. Requests to the neighborhood, sender pays.
+   The message being sent must include the Fee Attribute (#15) and the neighborhood MUST charge the sender the fee.
+// TODO: Requests to the neighborhood, receiver pays.
```
[the remark this answers is not in the document]
Comment (hansl) [reply attributes/network/15_fee.adoc:42]: I'll remove it. We'll expand on fee table entries for those.

**Comment by hansl** [comment]
We should support this but at a later time when we have a better idea. Let's close it for now.

**Pull request closed without merging**